src/MuTalk-Model/MTAnalysis.class.st

```diff
@@ -505,10 +506,36 @@ MTAnalysis >> doNotStopOnErrorOrFail [
 	stopOnErrorOrFail := false
 ]
 
+{ #category : 'accessing' }
+MTAnalysis >> filteredTests [
+
+	^ filteredTests
+]
+
+{ #category : 'accessing' }
+MTAnalysis >> filteredTests: anObject [
+
+	filteredTests := anObject
+]
+
+{ #category : 'as yet unclassified' }
+MTAnalysis >> filteredTestsFrom: aTestCollection [
+
+	filteredTests := aTestCollection
+		                 inject: OrderedCollection new
+		                 into: [ :accumulator :test |
+			                 accumulator add:
+				                 test -> testFilter filteredTestReason.
+			                 accumulator ]
```

Comment:
Hmm this method is strange :).

- First, this inject:into: could be replaced by a `collect:` (a simple `map`) no?
- Second, the objective of this method is to store the tests + reason, right?
I have one question here. All filtered tests have the same reason right now, right? I wonder why the test filter does not return the "tuple" already instead of creating it in here... It's the filter that decides to filter, then it could be the filter that attaches this to the test :)

Reply:
You're right, I refactored the test filters in this direction